.github/workflows/integration-tests.yml

```diff
@@ -199,22 +203,30 @@ jobs:
         # "restore" step.  This is to prevent the caches from accumulating stale
         # files over time.
         name: Restore cache of ccache and Triton compilation artifacts
-        if: github.event_name != 'push'
+        id: restore-build-cache
+        if: github.ref != 'refs/heads/main'
         uses: actions/cache/restore@v4
         with:
           path: |
             ~/.triton/cache
-            ~/.cache/ccache
+            ~/.ccache
           # Restore the most recent cache entry.
-          restore-keys: triton-artifacts-${{ runner.os }}-${{ runner.arch }}-${{ runner.name }}-llvm-${{ steps.cache-key.outputs.llvm }}-
+          restore-keys: |
+            triton-artifacts-${{ runner.os }}-${{ runner.arch }}-${{ env.RUNNER_TYPE }}-llvm-${{ steps.cache-key.outputs.llvm }}-
+            triton-artifacts-${{ runner.os }}-${{ runner.arch }}-${{ env.RUNNER_TYPE }}-
           # We expect this cache key never to hit and for us to fall back
           # unconditionally to the restore-key, so it doesn't actually matter
           # what we put here (so long as it doesn't hit an existing key).
-          key: triton-artifacts-${{ runner.os }}-${{ runner.arch }}-${{ runner.name }}-llvm-${{ steps.cache-key.outputs.llvm }}-${{ steps.cache-key.outputs.datetime }}
-      - name: Inspect cache directory
+          key: triton-artifacts-${{ runner.os }}-${{ runner.arch }}-${{ env.RUNNER_TYPE }}-llvm-${{ steps.cache-key.outputs.llvm }}-${{ steps.cache-key.outputs.datetime }}
+      - name: Inspect cache directories
         run: |
           mkdir -p ~/.triton
           ls -alh ~/.triton
+          du -sh ~/.triton/**
+
+          mkdir -p ~/.ccache
```

Comment:
The default cache directory if `~/.ccache` doesn't exist is actually different from OS to OS, so the macOS build was looking in the wrong directory.

Reply:
You meant other platforms use `~/.triton` but macos uses `~/.ccache`?

Reply:
I'm referring to the ccache directory. Linux defaults to `~/.cache/ccache` whereas macOS defaults to `~/Library/Caches/ccache\

Reply:
I'm confused, you're using `~/.ccache`, right?

Reply:
Yes, ccache has multiple defaults which have an order of precedence. `~/.ccache` is the highest precedence and the most cross-platform. Then if that doesn't already exist it will fall back to the platform specific options. See https://ccache.dev/manual/latest.html#_configuration_options in the cache_dir section

Reply:
OK, I get it now. Thanks for the reference